pkg/rotator/rotator.go

```diff
@@ -692,6 +696,12 @@ func (r *ReconcileWH) Reconcile(ctx context.Context, request reconcile.Request)
 	}
 
 	if secret.GetDeletionTimestamp().IsZero() {
+		if r.refreshCertIfNeededDelegate != nil {
```

Comment:
We should return if the cert was rotated, otherwise we are reconciling with a stale secret. That will probably require changing the signature of refreshCertIfNeeded

Reply:
I added a `return reconcile.Result{Requeue: true}, nil` block now 💯